### PR TITLE
Code is written, tests are not passing. Need to refactor after lab

### PR DESCRIPTION
**Pull request by sajafisher29**
Need to move on to labwork. Will come back to the tests and the README asap.

- [~] Top-level README “Table of Contents” is updated
 - [x] Feature tasks for this challenge are completed
 - [~] Unit tests written and passing
     - [x] “Happy Path” - Expected outcome
     - [x] Expected failure
     - [x] Edge Case (if applicable/obvious)
 - [~] README for this challenge is complete
     - [ ] Summary, Description, Approach & Efficiency, Solution
     - [ ] Link to code
     - [x] Picture of whiteboard

### CODE REVIEW

File: code401challenges/src/main/java/linkedList/LinkedList.java [hunk 1]
```diff
@@ -50,12 +50,82 @@ public String toString() {
...
+        Node<T> currentNode = this.head;
+
+        //If there are no nodes in the list
+        if (currentNode == null) {
+            Node newNode = new Node(value);
+            currentNode = newNode;
+        }
+
+        //Traverse to the end of the list and then append
+        while (currentNode.next != null) {
```
Comment (ncarignan) [inline code401challenges/src/main/java/linkedList/LinkedList.java]: this while loop is an infinite loop. your condition checks if current.next != null, within the loop if current.next becomes null, you add a node to the end of the list, the while loop reiterates and the same thing happens. you should move the logic of adding a new Node out of the if statement and put it after the while loop

File: code401challenges/src/main/java/linkedList/LinkedList.java [hunk 2]
```diff
@@ -50,12 +50,82 @@ public String toString() {
...
+        if (currentNode == null) {
+            return false;
+        } else if (currentNode.value == value) {
+            Node newNode = new Node(newValue, head);
+            head = newNode;
+            return true;
+        }
+
+        //Traverse the list to look for the node with the desired value
+        while (currentNode.next.value != value && currentNode.next != null) {
```
Comment (ncarignan) [inline code401challenges/src/main/java/linkedList/LinkedList.java]: you are checking currentNode.next.value before you find if currentNode.next is null. This throws a null pointer exception if next is null and you access its value property

File: code401challenges/src/main/java/linkedList/LinkedList.java (hunk 1 above)
Comment (sajafisher29) [reply code401challenges/src/main/java/linkedList/LinkedList.java]: Going back to this it looks like I caught it since submitting. The updated code is below:
while (currentNode.next != null) {
                currentNode = currentNode.next;
            }
            currentNode.next = new Node(value);

File: code401challenges/src/main/java/linkedList/LinkedList.java (hunk 2 above)
Comment (sajafisher29) [reply code401challenges/src/main/java/linkedList/LinkedList.java]: Looks like I refactored this one as well.
while (currentNode != null) {
            if (currentNode.value == desiredValue) {
                return true;
            }
            currentNode = currentNode.next;
        }
        return false;
